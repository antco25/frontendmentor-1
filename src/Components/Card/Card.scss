@import '../Utils/common';

.card {
    width: 100%;
    display: inline-block;
    border-radius: 0.6rem;
    box-shadow: 0 0.5rem $neutral-3;
    padding: 1.2rem;
    margin-bottom: 2rem;
    text-align: center;

    &.card-default {
        background-color: $primary-0;
        color: black;
    }

    &.card-accent {
        background-color: $accent-1;
    }
}
