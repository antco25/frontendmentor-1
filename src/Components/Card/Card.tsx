import { FunctionComponent } from 'react';
import './Card.scss';

type CardType = "default" | "accent";

type CardProps = {
    type? : CardType
}

function getClassName(type? : CardType) {
    if (type === "accent")
        return "card card-accent"
    else
        return "card card-default";
}

const Card : FunctionComponent<CardProps> = ({type, children}) => {

    const className = getClassName(type);

    return (
        <div className={className}>
            {children}
        </div>
    )
}

export default Card;