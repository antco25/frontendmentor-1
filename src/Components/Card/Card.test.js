import render from 'react-test-renderer'
import Card from './Card'

it('renders correctly', () => {
  const tree = render
    .create(<Card />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with type "default"', () => {
  const tree = render
    .create(<Card type="default" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with type "accent"', () => {
  const tree = render
    .create(<Card type="accent" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});