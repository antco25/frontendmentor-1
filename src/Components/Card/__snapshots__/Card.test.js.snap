// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card card-default"
/>
`;

exports[`renders correctly with type "default" 1`] = `
<div
  className="card card-default"
/>
`;

exports[`renders correctly with type "accent" 1`] = `
<div
  className="card card-accent"
/>
`;


